packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -460,84 +460,72 @@ class IrisGridTableModel extends IrisGridModel {
     return '';
   }
 
-  getColumnsWithLayoutHints = memoize(
-    (columns, hints, currentFrozenColumns) => {
-      if (hints || currentFrozenColumns.length) {
-        const columnMap = new Map();
-        columns.forEach(col => columnMap.set(col.name, col));
+  getColumnsWithLayoutHints = memoize((columns, hints, userFrozenColumns) => {
+    if (hints || userFrozenColumns.length) {
+      const columnMap = new Map();
+      columns.forEach(col => columnMap.set(col.name, col));
 
-        let frontColumns = [];
-        let backColumns = [];
-        let frozenColumns = [];
+      let frontColumns = [];
+      let backColumns = [];
+      let frozenColumns = [];
 
-        if (hints?.frontColumns) {
-          frontColumns = hints.frontColumns
-            .map(name => columnMap.get(name))
-            .filter(Boolean);
-        }
-        if (hints?.backColumns) {
-          backColumns = hints.backColumns
-            .map(name => columnMap.get(name))
-            .filter(Boolean);
-        }
-        if (hints?.frozenColumns) {
-          frozenColumns = hints.frozenColumns
-            .map(name => columnMap.get(name))
-            .filter(Boolean);
-        }
-
-        if (
-          frontColumns.length !== (hints?.frontColumns?.length ?? 0) ||
-          backColumns.length !== (hints?.backColumns?.length ?? 0) ||
-          frozenColumns.length !== (hints?.frozenColumns?.length ?? 0)
-        ) {
-          throw new Error(
-            'Layout hints are invalid (contain invalid column names)'
-          );
-        }
-
-        // Handle frozen columns by user UI action.
-        if (currentFrozenColumns.length) {
-          frozenColumns = [
-            ...frozenColumns,
-            ...currentFrozenColumns
-              .map(name => columnMap.get(name))
-              .filter(Boolean),
-          ];
-        }
+      if (hints?.frontColumns) {
+        frontColumns = hints.frontColumns
+          .map(name => columnMap.get(name))
+          .filter(Boolean);
+      }
+      if (hints?.backColumns) {
+        backColumns = hints.backColumns
+          .map(name => columnMap.get(name))
+          .filter(Boolean);
+      }
+      if (userFrozenColumns.length) {
+        frozenColumns = userFrozenColumns
+          .map(name => columnMap.get(name))
+          .filter(Boolean);
+      }
```

Comment:
If userFrozenColumns isn't defined, then you should be looking at the layout hints frozen columns